docs/data/material/components/app-bar/ElevateAppBar.tsx

```diff
@@ -13,7 +13,7 @@ interface Props {
    * You won't need it on your project.
    */
   window?: () => Window;
-  children?: React.ReactElement<any>;
+  children?: React.ReactElement<unknown & { elevation?: number }>;
```

Comment:
Here `elevation` prop is added to children, since `unknown` type doesn't have `elevation` type, i explicitly added. I'm not entirely sure if this is optimal, but please let me know if you have better suggestion. https://github.com/mui/material-ui/blob/6019f2ae8266892d0209f29b54666a683df6014a/docs/data/material/components/app-bar/ElevateAppBar.tsx#L32

Reply:
I think we can have only the elevation prop:

```suggestion
  children?: React.ReactElement<{ elevation?: number }>;
```

As unknown is absorbed in intersections: 

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type:~:text=//%20In%20an%20intersection%20everything%20absorbs%20unknown

Reply:
fixed here https://github.com/mui/material-ui/pull/43402/commits/3ff9828a85531cb6825444e933571ba2d7d20fef